Source/WebCore/editing/MarkupAccumulator.cpp

```diff
@@ -423,7 +424,13 @@ void MarkupAccumulator::startAppendingNode(const Node& node, Namespaces* namespa
             m_markup.append(" shadowrootserializable=\"\""_s);
         if (shadowRoot->isClonable())
             m_markup.append(" shadowrootclonable=\"\""_s);
-        if (shadowRoot->protectedHost()->customElementRegistry() != shadowRoot->registryForBindings())
+        bool shouldAppendRegistryAttribute = [&] {
+            RefPtr documentRegistry = shadowRoot->document().customElementRegistryForBindings();
```

Comment:
We don't wanna call this. This will create custom element registry for the document if there isn't one.